### PR TITLE
add separate server info file paths for services on client side

### PR DESCRIPTION
**Pull request by a3hadi**
Implement #1456 on client side

### CODE REVIEW

**Comment by KeranYang** [comment]
@a3hadi , with this change, the platform stops supporting all the old UDF images (hence all e2e fail.). Users have to re-build UDF/Source/Sink images using the latest SDKs. Since server info hasn't been widely used, let's see if we can update on the platform side in a way that when a server info file is not found because of incompatible file path, the platform numa container can decide to ignore it and continue processing.

**Comment by KeranYang** [comment]
@a3hadi , could you also look into why the udsource e2e succeeded? Thanks!

**Comment by a3hadi** [comment]
> @a3hadi , with this change, the platform stops supporting all the old UDF images (hence all e2e fail.). Users have to re-build UDF/Source/Sink images using the latest SDKs. Since server info hasn't been widely used, let's see if we can update on the platform side in a way that when a server info file is not found because of incompatible file path, the platform numa container can decide to ignore it and continue processing.

I think ignoring incompatible server info path, with a message to remind to upgrade sdk is a good idea (since using the new server info paths isn't mission critical to continue processing anyways). I will focus on getting the current tests to pass first and then look into this.

**Comment by a3hadi** [comment]
After discussing with Keran, I think the best course of action for the "error handling" is simply, if the server info file path is not found, set it to the default path of `/var/run/numaflow/server-info`. This is because, the server info file will always be created. However there are two possible paths that it can have. If an old sdk version is being used it will be the aforementioned default path, if the sdk version is up to date it will be using the new isolated paths for each service. Thus falling back to the default path when the server info file path is not found will accommodate older sdk versions.

File: pkg/sdkclient/mapper/client.go [hunk 1]
```diff
@@ -36,7 +37,7 @@ type client struct {
 
 // New creates a new client object.
 func New(serverInfo *info.ServerInfo, inputOptions ...sdkclient.Option) (Client, error) {
-	var opts = sdkclient.DefaultOptions(sdkclient.MapAddr)
+	var opts = sdkclient.DefaultOptions(sdkclient.MapAddr, sdkserverinfo.MapServerInfoFile)
```
[the remark this answers is not in the document]
Comment (KeranYang) [reply pkg/sdkclient/mapper/client.go:40]: Please also learn about the Go option pattern and see if we are using it accurately here.

File: pkg/sdkclient/sideinput/client.go [hunk 2]
```diff
@@ -21,8 +21,6 @@ type client struct {
 	grpcClt sideinputpb.SideInputClient
 }
 
-var _ Client = (*client)(nil)
```
Comment (KeranYang) [inline pkg/sdkclient/sideinput/client.go:24]: Please keep this line. https://github.com/KeranYang/numaflow-notes/blob/main/GoFundamental.md#2-a-trick-to-ensure-an-object-implements-an-interface

File: pkg/sdkclient/sinker/client.go [hunk 3]
```diff
@@ -35,23 +35,19 @@ type client struct {
 	grpcClt sinkpb.SinkClient
 }
 
-var _ Client = (*client)(nil)
```
Comment (KeranYang) [inline pkg/sdkclient/sinker/client.go:38]: please add back

File: pkg/sdkclient/source/client.go [hunk 4]
```diff
@@ -36,26 +36,20 @@ type client struct {
 	grpcClt sourcepb.SourceClient
 }
 
-var _ Client = (*client)(nil)
```
Comment (KeranYang) [inline pkg/sdkclient/source/client.go:39]: please add back

**Review by KeranYang — APPROVED** [review]
LGTM. Please add back the interface check and then we can merge this in.

**Pull request merged**